packages/react/src/components/header/components/headerLanguageSelector/HeaderLanguageSelector.tsx

```diff
@@ -128,21 +118,21 @@ export const HeaderLanguageSelectorConsumer = ({
     <div className={classNames(classes.languageSelector, { [classes.fullWidthForMobile]: fullWidthForMobile })}>
       <div className={classNames(classes.languageNodes)}>{primaryLanguageNodes}</div>
       {(hasChildren || hasSecondaryLanguages) && (
-        <HeaderActionBarItemWithDropdown
+        <HeaderActionBarItem
           id="language-selection-more"
+          className={classes.languageSelectorActionBarItem}
           iconClassName={classes.languageSelectorDropdownIcon}
           dropdownClassName={classes.languageSelectorDropdown}
-          label={<IconAngleDown aria-hidden />}
-          closeLabel={<IconAngleUp aria-hidden />}
           icon={<IconGlobe aria-hidden />}
           closeIcon={<IconGlobe aria-hidden />}
           fullWidth={isNotLargeScreen}
           ariaLabel={ariaLabel}
           labelOnRight
         >
-          <SecondaryLanguages />
+          {secondaryLanguages?.length && languageHeading && <HeaderActionBarSubItem label={languageHeading} heading />}
+          {secondaryLanguages?.length && secondaryLanguages.map((node) => renderLanguageNode(node, false))}
```

Comment:
Hmm... this you could shorten just as `secondaryLanguages?.map` (leaving out the .length && check first)